covalent/executor/base.py

```diff
@@ -202,6 +206,31 @@ def __init__(
     ) -> None:
         super().__init__(*args, **kwargs)
 
+    def _init_runtime(self, loop=None, cancel_pool=None):
+        self._send_queue = asyncio.Queue()
+        self._recv_queue = queue.Queue()
+        self._loop = loop
+        self._cancel_pool = cancel_pool
+
+    def _notify(self, action: str, body: Any = None):
+        self._loop.call_soon_threadsafe(self._send_queue.put_nowait, (action, body))
+
+    def _notify_sync(self, action: str, body: Any = None):
+        self._notify(action, body)
+        return self._wait_for_response()
+
+    def _wait_for_response(self, timeout: int = 5):
+        status, body = self._recv_queue.get(timeout=timeout)
+        if status is None:
+            raise RuntimeError("Error waiting for response")
+        return body
+
+    def get_cancel_requested(self):
+        return self._notify_sync("get", "cancel_requested")
```

Comment:
@venkatBala let's try to use constants or enums here instead of string literals

Reply:
@AlejandroEsquivel I understand... but the implementation has been as such that the threads on the receiving end of the queue are expecting these string literals to process them in task cancellation